File: ts-sdk/src/utils/get.chain.id.ts
import { JsonRpcProvider } from "ethers";
import { WithRetry } from "./with.retry";
import { assertExists } from "./assert";

// returns the chain id as reported by the "eth_chainId" json-rpc method
export type GetChainId = (provider: JsonRpcProvider) => Promise<number>;

export function provideGetChainId(withRetry: WithRetry): GetChainId {
  assertExists(withRetry, "withRetry");

  return async function getChainId(provider: JsonRpcProvider) {
    const chainId: string = await withRetry(provider.send.bind(provider), [
      "eth_chainId",
      [],
    ]);
    return parseInt(chainId);
  };
}

File: py-sdk/src/forta_bot/utils/get_chain_id.py
from typing import Callable
from web3 import AsyncWeb3
from .with_retry import WithRetry
from .assertions import assert_exists

# returns the chain id from the "eth_chainId" json-rpc method
GetChainId = Callable[[AsyncWeb3], int]


def provide_get_chain_id(with_retry: WithRetry) -> GetChainId:
    assert_exists(with_retry, "with_retry")

    async def get_chain_id(provider: AsyncWeb3) -> int:
        response = await with_retry(provider.provider.make_request, "eth_chainId", [])
        chain_id = int(response['result'], 0)
        return chain_id

    return get_chain_id
